### PR TITLE
Add support for OpenAI API, improve diagram generation, and fix route…

### PR DESCRIPTION
**Pull request by helioLJ**
# Add support for OpenAI API and improve diagram generation

This PR adds support for using the OpenAI API alongside the existing Anthropic API and improves the diagram generation process. It also notes a discovered bug that needs attention.

## Features
- Added option to choose between OpenAI and Anthropic APIs
- Improved Mermaid diagram generation and syntax fixing
- Enhanced error handling and user feedback

### fences/agent.py
- Added OpenAI import and integration
- Refactored `Agent` class to support both OpenAI and Anthropic
- Improved diagram generation with better syntax and error handling
- Added `fix_diagram_syntax` method to clean up generated diagrams

### fences/cli.py
- Removed hardcoded Anthropic API key prompt
- Updated to use new `Agent` initialization
- Improved diagram parsing and display

### pyproject.toml
- Added dependencies for `python-dotenv` and `openai`

## Benefits
- Users can now choose between OpenAI and Anthropic APIs
- Improved diagram quality and consistency
- Better error handling and user feedback during the process

## Known Issue (Not Fixed in this PR)
While testing, I discovered a bug when trying to click on a route node in the diagram. An image of the error has been attached to this PR for reference. This issue has not been addressed in the current changes and requires further investigation.
image

## Testing Limitations
Due to not having an Anthropic API key, I was unable to test the official version available via pip. So I don't if I just caused this bug or if it's happening in the official version also.

## Testing Checklist
Please test the following scenarios:
1. Using the OpenAI API
2. Using the Anthropic API (if possible)
3. Generating diagrams with various OpenAPI specifications
4. Verify the known issue with clicking route nodes in the generated diagram

## Additional Notes
- The attached error image shows the issue encountered when clicking route nodes (not fixed in this PR)
- Anthropic API functionality needs verification by someone with an active API key

Let me know if you need any clarifications or have suggestions for improvement.

### CODE REVIEW

**Comment by helioLJ** [comment]
Just to clarify, this PR should not be merged in the master branch before testing this node click bug!

File: fences/agent.py [hunk 1]
```diff
@@ -32,73 +39,131 @@
...
 class Agent():
-  def __init__(self, api_key):
-    self.initialize_agent(api_key)
+    def __init__(self):
+        self.initialize_agent()
 
-  def initialize_agent(self, api_key):
-    self.client = anthropic.Anthropic(api_key=api_key)
+    def initialize_agent(self):
+        api_choice = input("Do you want to use OpenAI or Anthropic? (openai/anthropic): ").lower()
```
Comment (LTMenezes) [inline fences/agent.py]: I want to keep this outside of the agent so all the interactions with the command line are organised under the cli.py. This way agent.py can be used as a standalone library in the future.

Can you move all the inputs and prints to console back to an argument to the constructor?

**Review by LTMenezes — CHANGES_REQUESTED** [review]

**Comment by LTMenezes** [comment]
This is so good @helioLJ, thanks man!
Let me know if you need any help doing this change that I asked, I want to merge this asap

File: fences/agent.py (hunk 1 above)
Comment (helioLJ) [reply fences/agent.py]: Sure, it makes all sense, just commited it!

**Comment by helioLJ** [comment]
> This is so good @helioLJ, thanks man! Let me know if you need any help doing this change that I asked, I want to merge this asap

I just commited that, if you think it still needs any change, let me know

**Comment by LTMenezes** [comment]
Merged, thanks for the PR @helioLJ !

**Comment by helioLJ** [comment]
> Merged, thanks for the PR @helioLJ !

Ty for your review, nice project!

**Pull request merged**